File: app/src/main/java/android/hardware/fingerprint/MyAuthenticationResult.java
package android.hardware.fingerprint;

public class MyAuthenticationResult extends FingerprintManager.AuthenticationResult {

    FingerprintManager.CryptoObject cryptoObject;

    public void setCryptoObject(FingerprintManager.CryptoObject cryptoObject) {
        this.cryptoObject = cryptoObject;
    }

    @Override
    public FingerprintManager.CryptoObject getCryptoObject() {
        if (cryptoObject != null) {
            return cryptoObject;
        }
        return super.getCryptoObject();
    }
}

File: app/src/main/java/android/hardware/fingerprint/MyAuthenticationResult.kt
@file:Suppress("DEPRECATION")

package android.hardware.fingerprint

import android.hardware.biometrics.BiometricPrompt

class MyAuthenticationResult(private val result: BiometricPrompt.AuthenticationResult) :
    FingerprintManager.AuthenticationResult() {

    private fun BiometricPrompt.CryptoObject.toLegacy() = when {
        cipher != null -> FingerprintManager.CryptoObject(cipher)
        mac != null -> FingerprintManager.CryptoObject(mac)
        signature != null -> FingerprintManager.CryptoObject(signature)
        else -> null
    }

    override fun getCryptoObject(): FingerprintManager.CryptoObject? {
        return result.cryptoObject?.toLegacy()
    }
}
